### PR TITLE
ShapeType::Type -> ShapeType + Interpolated<ShapeType>

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/base/anim/interpolated.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 #include "base/conv/parse.h"
 #include "base/conv/tostring.h"
+#include "base/math/fuzzybool.h"
```
Comment (simzer) [inline src/base/anim/interpolated.h:12]: Do we need this here after all?

File: src/apps/weblib/js-api/vizzu.js [hunk 2]
```diff
@@ -71,10 +71,14 @@ export default class Vizzu {
       try {
         return f(...params);
       } catch (e) {
-        let address = parseInt(e, 10);
-        let cMessage = this.module._vizzu_errorMessage(address);
-        let message = this.module.UTF8ToString(cMessage);
-        throw new Error("error: " + message);
+        if (Number.isInteger(e)) {
```
Comment (simzer) [inline src/apps/weblib/js-api/vizzu.js:74]: Please put a line about this in the Changelog's ureleased/fixed section

File: src/chart/generator/guides.cpp [hunk 3]
```diff
@@ -31,14 +31,14 @@ bool GuidesByAxis::operator==(const GuidesByAxis &other) const
 void Guides::init(const Axises &axises, const Options &options)
 {
 	auto isCircle =
-	    options.shapeType.getFactor(ShapeType::Type::circle);
-	auto isLine = options.shapeType.getFactor(ShapeType::Type::line);
-	auto isHorizontal = options.horizontal;
+	    options.shapeType.factor<Math::FuzzyBool>(ShapeType::circle) != false;
```
Comment (simzer) [inline src/chart/generator/guides.cpp:34]: options.shapeType.get() != ShapeType::circle

Comment (simzer) [reply src/chart/generator/guides.cpp:34]: In this function, nothing got interpolated yet, FuzzyBool::operator bool() and Interpolated::get() can be called safely.

File: src/chart/options/shapetype.h [hunk 4]
```diff
@@ -12,81 +12,11 @@ namespace Vizzu
...
-
-private:
-	ShapeFactors shapeFactors;
-};
+enum class ShapeType : uint32_t { rectangle, circle, area, line };
 
-std::string toString(ShapeType::Type type);
-bool canOverlap(ShapeType::Type type);
-bool isConnecting(ShapeType::Type type);
+std::string toString(ShapeType type);
```
Comment (simzer) [inline src/chart/options/shapetype.h]: This function seems to be deleted.

File: src/base/anim/interpolated.h (hunk 1 above)
Comment (schaumb) [reply src/base/anim/interpolated.h:12]: no

File: src/chart/generator/guides.cpp (hunk 3 above)
Comment (schaumb) [reply src/chart/generator/guides.cpp]: done

File: src/chart/options/shapetype.h (hunk 4 above)
Comment (schaumb) [reply src/chart/options/shapetype.h]: deleted

File: src/apps/weblib/js-api/vizzu.js (hunk 2 above)
Comment (schaumb) [reply src/apps/weblib/js-api/vizzu.js:74]: Is it good? https://github.com/vizzuhq/vizzu-lib/pull/224/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR19

**Review by simzer — APPROVED** [review]

**Review by simzer — DISMISSED** [review]

**Review by simzer — APPROVED** [review]

**Pull request merged**